src/components/hoc/RestrictedRoute.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router';
+
+export default function RestrictedRoute({
+  isAllowed = false,
+  component: Component,
+  render: RenderedComponent,
+  redirectTo,
+  ...routeProps
+}) {
+  return (
+    <Route
+      {...routeProps}
+      render={(props) => {
+        if (isAllowed) {
+          // Compatibility with both Route render or Route component
+          if (!!RenderedComponent) {
```

Comment:
shouldn't this use render instead of RenderedComponent, considering render is the key?

Reply:
The destructuring syntax is renaming the prop so that it is not confused with the above `render` prop. Purely there for readability.